ggml.c

```diff
@@ -287,49 +287,51 @@ void ggml_time_init(void) {
     QueryPerformanceFrequency(&frequency);
     timer_freq = frequency.QuadPart;
 }
-int64_t ggml_time_ms(void) {
+
+int64_t ggml_real_time_ms(void) {
     LARGE_INTEGER t;
     QueryPerformanceCounter(&t);
     return (t.QuadPart * 1000) / timer_freq;
 }
-int64_t ggml_time_us(void) {
+
+int64_t ggml_real_time_us(void) {
     LARGE_INTEGER t;
     QueryPerformanceCounter(&t);
     return (t.QuadPart * 1000000) / timer_freq;
 }
+
+int64_t ggml_process_time_ms(void) {
+    return (clock() * 1000) / CLOCKS_PER_SEC;
+}
+
+int64_t ggml_process_time_us(void) {
+    return (clock() * 1000000) / CLOCKS_PER_SEC;
+}
 #else
 void ggml_time_init(void) {}
-int64_t ggml_time_ms(void) {
+int64_t ggml_real_time_ms(void) {
     struct timespec ts;
     clock_gettime(CLOCK_MONOTONIC, &ts);
     return (int64_t)ts.tv_sec*1000 + (int64_t)ts.tv_nsec/1000000;
 }
 
-int64_t ggml_time_us(void) {
+int64_t ggml_real_time_us(void) {
     struct timespec ts;
     clock_gettime(CLOCK_MONOTONIC, &ts);
     return (int64_t)ts.tv_sec*1000000 + (int64_t)ts.tv_nsec/1000;
 }
-#endif
 
-int64_t ggml_cycles(void) {
-    return clock();
+int64_t ggml_process_time_ms(void) {
+    struct timespec ts;
+    clock_gettime(CLOCK_PROCESS_CPUTIME_ID, &ts);
+    return (int64_t)ts.tv_sec*1000 + (int64_t)ts.tv_nsec/1000000;
 }
 
-int64_t ggml_cycles_per_ms(void) {
-    return CLOCKS_PER_SEC/1000;
+int64_t ggml_process_time_us(void) {
+    struct timespec ts;
+    clock_gettime(CLOCK_PROCESS_CPUTIME_ID, &ts);
+    return (int64_t)ts.tv_sec*1000000 + (int64_t)ts.tv_nsec/1000;
 }
-
-#ifdef GGML_PERF
-#define ggml_perf_time_ms()       ggml_time_ms()
-#define ggml_perf_time_us()       ggml_time_us()
-#define ggml_perf_cycles()        ggml_cycles()
-#define ggml_perf_cycles_per_ms() ggml_cycles_per_ms()
-#else
-#define ggml_perf_time_ms()       0
-#define ggml_perf_time_us()       0
-#define ggml_perf_cycles()        0
-#define ggml_perf_cycles_per_ms() 0
```

Comment:
This is actually needed and should not be removed.

The idea is to use the `ggml_perf_...` macros in `ggml.c` in order to time the performance of critical sections like `ggml_compute_forward_...` and others. However, since these sections can be invoked many hundreds of times and the clock system calls are not infinitely cheap, we want to be able to disable these measurements.

This is what the `GGML_PERF` flag does - when we are developing and measuring the performance, we define `GGML_PERF` and use `ggml_graph_print()` to see timing results for various `ggml` operations. However, when we are ready, we want to disable `GGML_PERF` in order to make all clock calls noop.

The basic rule is that all clock calls inside the `ggml.c` source file should be done via the `ggml_perf_...` macros in order to be able to disable them.

Reply:
Makes sense. I've added it back, but removed the cycle and millisecond related macros as they aren't used anywhere (cycles having been replaced with process time).